docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -288,6 +296,28 @@ public String getEnvironmentsString() {
         return Joiner.on("\n").join(environment);
     }
 
+    @CheckForNull
+    public List<String> getExtraHosts() {
+        return extraHosts;
+    }
+
+    public void setExtraHosts(List<String> extraHosts) {
+        this.extraHosts = extraHosts;
+    }
+
+    @DataBoundSetter
+    public void setExtraHostsString(String extraHostsString) {
+        setExtraHosts(splitAndFilterEmptyList(extraHostsString, "\n"));
+    }
+
+    public String getExtraHostsString() {
+        if (getExtraHosts() == null) {
```

Comment:
Not really critical. This value is used in UI, probably it can be null, but 'no configuration' from ui is returned as empty string, so emulating the same behaviour. I can try return null if you want.